src/payment/strategies/square/square-payment-strategy.ts

```diff
@@ -281,9 +286,65 @@ export default class SquarePaymentStrategy implements PaymentStrategy {
             masterpassBtn.style.display = 'inline-block';
         }
     }
-}
 
-export interface DeferredPromise {
-    resolve(resolution?: NonceInstrument): void;
-    reject(reason?: any): void;
+    private _getBillingContact(): Contact {
+        const state = this._store.getState();
+        const billingAddress = state.billingAddress.getBillingAddressOrThrow();
+
+        return {
+            givenName: billingAddress.firstName,
+            familyName: billingAddress.lastName,
+            email: billingAddress.email || '',
+            country: billingAddress.countryCode,
+            countryName: billingAddress.country,
+            region: '',
+            city: billingAddress.city,
+            postalCode: billingAddress.postalCode,
+            addressLines: [ billingAddress.address1, billingAddress.address2],
+            phone: billingAddress.phone,
+        };
+    }
+
+    private _getAmountAndCurrencyCode(): string[] {
+        const state = this._store.getState();
+        const storeConfig = state.config.getStoreConfigOrThrow();
+        const checkout = state.checkout.getCheckoutOrThrow();
+
+        return [String(checkout.grandTotal), storeConfig.currency.code];
```

Comment:
Shouldn't the currency code be fetched from Cart config instead of store config?